arangod/VocBase/Methods/Databases.cpp

```diff
@@ -402,15 +404,24 @@ Result Databases::create(ArangodServer& server, ExecContext const& exec,
       return res;
     }
 
-    if (createInfo.replicationVersion() == replication::Version::TWO &&
-        !replication2::EnableReplication2) {
-      using namespace std::string_view_literals;
-      auto const message =
-          R"(Replication version 2 is disabled in this binary, but trying to create a version 2 database.)"sv;
-      LOG_TOPIC("e768d", ERR, Logger::REPLICATION2) << message;
-      // Should not happen during testing
-      TRI_ASSERT(false);
-      return Result(TRI_ERROR_NOT_IMPLEMENTED, message);
+    if (createInfo.replicationVersion() == replication::Version::TWO) {
+      if (!replication2::EnableReplication2) {
+        using namespace std::string_view_literals;
+        auto const message =
+            R"(Replication version 2 is disabled in this binary, but trying to create a version 2 database.)"sv;
+        LOG_TOPIC("e768d", ERR, Logger::REPLICATION2) << message;
+        // Should not happen during testing
+        TRI_ASSERT(false);
+        return Result(TRI_ERROR_NOT_IMPLEMENTED, message);
+      }
+
+      auto& selector = server.getFeature<EngineSelectorFeature>();
+      auto& engine = selector.engine<RocksDBEngine>();
+      if (engine.syncThread() == nullptr) {
+        return Result(TRI_ERROR_ILLEGAL_OPTION,
+                      "Unless using the custom WAL, automatic syncing must be "
```

Comment:
Please not name the Custom WAL in this publicly visible message.
This is nothing our customers can control.

I would prefer a message that helps the user, so it should in my opinion contain info on the setting to avoid this situation
(taken from above:)
"--rocksdb.sync-interval"